sdk/src/version/version.cc

```diff
@@ -0,0 +1,24 @@
+// Please DONOT touch this file.
+// Any changes done here would be overwritten by pre-commit git hook
```

Comment:
Updating version via githook for each commit/merge might break forked repo? Is it better to just generated the version info and build it into the library in our build script, but don't commit it?

Reply:
Yes valid point. Things would be much simpler if we decide not to commit the version information. We can just call the `buildscripts/pre-commit` script created above as part of GitHub release action as I mentioned in the PR description. The problem with adding it as part of build script ( or cmake / bazel ) is that we won't be doing build as part of release  ( only source is released ). So we can create wrap release process through  github action(s) which would actually run the `buildscripts/pre-commit` first to create version.cc file and then release the source. 

Also agree on issues of adding it as pre-commit hook. It shouldn't be enforced. And the release manager should be doing this step ( i.e. copying `buildscripts/pre-commit` to `.git/hooks/pre-commit`) before doing release process. Anyway not something recommended. I will remove the comment about `pre-commit` hook from above with warning message that this file is updated automatically through script.

Reply:
Have updated the comment.